cfg/17.06/definitions.yaml

```diff
@@ -859,12 +859,18 @@ groups:
     description: "Ensure a user for the container has been created (Scored)"
     audit: "docker ps --quiet --all | xargs docker inspect --format '{{ .Id }}: User={{ .Config.User }}'"
     tests:
+      bin_op: and
```

Comment:
I just tried this out, and it returns output like this: 

e9645a9322ab3063b3e85c60a57a5ef8fa742a9bfa8b84faa447018375510de9: User=

I'm pretty sure in this example the container is running as root, and therefore the test should fail, but I'm not convinced that our check will find it. Should we really be testing for flag "User" and that it's not empty, and that it's not "root", and that it's not 1?

Reply:
I'm not going to insist on it, but if I were working on this I'd be inclined to add go unit tests with some sample data to confirm that the changes work as expected, especially in cases like this where it's not absolutely obvious what the output from the audit command will be.

Reply:
@lizrice ,
>I'm pretty sure in this example the container is running as root, and therefore the test should fail, but I'm not convinced that our check will find it.
Yes, the root flag is incorrect. I also wondered why is it specified "root" as flag. The regex will work correctly though, it will return "", and both checks will be false  as "" is contained in every string, so the !string.Contains() wll return false and "" not eq "" will return false. The check however will also fail when a user will be set. I think the checks need to be revisited. Will change the check flag to User. 

In case flag is User, the checks pass or fail as expected: https://play.golang.org/p/HvwadS87D-i

Unit tests will be added. Assigned Itai in bench-common.